src/logging_manager/mod.rs

```diff
@@ -0,0 +1,340 @@
+        error!("test error");
+        // "[YYYY-MM-DD HH:MM:SS] INFO  test info"
+        info!("test info");
+        // "[YYYY-MM-DD HH:MM:SS] TRACE test trace"
+        trace!("test trace");
+        // "[YYYY-MM-DD HH:MM:SS] WARN  test warning"
+        warn!("test warning");
+
+        // sleep because logging is done on a different thread (and will take time)
+        thread::sleep(time::Duration::from_millis(10));
+
+        // make sure file is there and contents are correct
+        assert!(
+            log_file_path.exists()
+        );
+        let log_file = File::open(log_file_path)?;
+        let mut lines = BufReader::new(log_file).lines();
+        assert!(lines.next().unwrap().unwrap().ends_with("] DEBUG test debug"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] ERROR test error"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] INFO  test info"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] TRACE test trace"));
+        assert!(lines.next().unwrap().unwrap().ends_with("] WARN  test warning"));
+        assert!(lines.next().is_none());
+
+        // clean up
+        remove_dir_all(logging_directory)?;
+        assert_eq!(false, logging_directory.exists());
+        Ok(())
+    }
+
+    #[test]
+    fn test_check_size() -> Result<(), std::io::Error> {
+
+        // random files and configs
+        let file1 = Path::new("2020_08_01.log");
+        let file2 = Path::new("2020_08_02.log");
+        let file3 = Path::new("2020_08_03.log");
+        let random_text = "Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.";
+        let config1 = LoggingConfig {
+            logging_directory: Path::new("./test_logs2/"),
+            max_dir_size: 0
+        };
+        let config2 = LoggingConfig {
+            logging_directory: Path::new("./test_logs3/"),
+            max_dir_size: 1000
+        };
+        let config3 = LoggingConfig {
+            logging_directory: Path::new("./test_logs4/"),
+            max_dir_size: 10000
+        };
+
+        // config 1: max dir size is 0, all files should be deleted
+        create_dir(&config1.logging_directory)?;
+        let mut file = File::create(&config1.logging_directory.join(&file1))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config1.logging_directory.join(&file2))?;
+        file.write_all(random_text.as_bytes())?;
+        file = File::create(&config1.logging_directory.join(&file3))?;
+        file.write_all(random_text.as_bytes())?;
+        file.sync_all()?;
+        check_size(&config1)?;
```

Comment:
I don't see any asserts after you call check_size. check_size should delete all the log files in this case right?

Reply:
rm_dir will panic if the directory is not empty, so that is my "assert" here

Reply:
Ah I see